hathor/indexes/base_index.py

```diff
@@ -29,12 +32,14 @@ class BaseIndex(ABC):
     created to generalize how we initialize indexes and keep track of which ones are up-to-date.
     """
 
-    def init_start(self, indexes_manager: 'IndexesManager') -> None:
+    @inlineCallbacks
+    def init_start(self, reactor: 'Reactor', indexes_manager: 'IndexesManager') -> Generator[Any, Any, None]:
```

Comment:
This method doesn't really need to use `@inlineCallback`, it can simply indicate that is an "async" method by using `-> Deferred` and the implementations can choose to use `@inlineCallback` to simplify making a function that returns a `Deferred`. Also it can always return `defer.succeed(None)` to have a similar effect and maintain the interface of returning a `Deferred`.

Reply:
Great tip. Thanks! Done!